pkg/genericapiserver/registry/generic/registry/store.go

```diff
@@ -50,94 +50,115 @@ import (
 
-// Store implements pkg/api/rest.StandardStorage.
-// It's intended to be embeddable, so that you can implement any
-// non-generic functions if needed.
-// You must supply a value for every field below before use; these are
-// left public as it's meant to be overridable if need be.
-// This object is intended to be copyable so that it can be used in
-// different ways but share the same underlying behavior.
+// Store implements pkg/api/rest.StandardStorage. It's intended to be
+// embeddable and allows the consumer to implement any non-generic functions
+// that are required. This object is intended to be copyable so that it can be
+// used in different ways but share the same underlying behavior.
+//
+// All fields are required unless specified.
 //
 // The intended use of this type is embedding within a Kind specific
-// RESTStorage implementation. This type provides CRUD semantics on
-// a Kubelike resource, handling details like conflict detection with
-// ResourceVersion and semantics. The RESTCreateStrategy and
-// RESTUpdateStrategy are generic across all backends, and encapsulate
-// logic specific to the API.
+// RESTStorage implementation. This type provides CRUD semantics on a Kubelike
+// resource, handling details like conflict detection with ResourceVersion and
+// semantics. The RESTCreateStrategy, RESTUpdateStrategy, and
+// RESTDeleteStrategy are generic across all backends, and encapsulate logic
+// specific to the API.
 //
 // TODO: make the default exposed methods exactly match a generic RESTStorage
 type Store struct {
-	// Called to make a new object, should return e.g., &api.Pod{}
+	// NewFunc returns a new object, e.g., &api.Pod{}
 	NewFunc func() runtime.Object
 
-	// Called to make a new listing object, should return e.g., &api.PodList{}
+	// NewListFunc returns a new list object for this resource, e.g., &api.PodList{}.
 	NewListFunc func() runtime.Object
 
-	// Used for error reporting
+	// QualifiedResource is the pluralized name of the resource.
 	QualifiedResource schema.GroupResource
 
-	// Used for listing/watching; should not include trailing "/"
+	// KeyRootFunc returns the root etcd key for this resource; should not
+	// include trailing "/".  This is used for operations that work on the
+	// entire collection (listing and watching).
+	//
+	// KeyRootFunc and KeyFunc must be supplied together or not at all.
 	KeyRootFunc func(ctx genericapirequest.Context) string
 
-	// Called for Create/Update/Get/Delete. Note that 'namespace' can be
-	// gotten from ctx.
+	// KeyFunc returns the key for a specific object in the collection.
+	// KeyFund is dalled for Create/Update/Get/Delete. Note that 'namespace'
+	// can be gotten from ctx.
+	//
+	// KeyFunc and KeyRootFunc must be supplied together or not at all.
 	KeyFunc func(ctx genericapirequest.Context, name string) (string, error)
 
-	// Called to get the name of an object
+	// ObjectNameFunc returns the name of an object or an error.
```

Comment:
It's actually the opposite of that -- it returns the etcd path, minus the name of the object.